Sources/Account/Balances.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+
+public struct Balances {
+    public let balances: [TokenId: Balance]
+    public var tokenIds: Set<TokenId> {
+        Set(balances.keys)
+    }
+    
+    var mobBalance: Balance {
+        guard let balance = balances[.MOB] else {
+            return Balance(
+                amountLow: 0,
+                amountHigh: 0,
+                blockCount: blockCount,
+                tokenId: .MOB)
+        }
+        return balance
+    }
+    
+    let blockCount: UInt64
+    
+    init(balances: [Balance], blockCount: UInt64) {
+        self.balances = balances.reduce(into: [TokenId:Balance](), { result, balance in
+            result[balance.tokenId] = balance
```

Comment:
Do you think it might be worth adding some error checking here to make sure we don't have input with more than one value for a particular token ID in the input array?

e.g. if we have two Balance values in the input balances, both for .Mob, then the second one will overwrite the first...

Just thinking it's a chance to add a check for bad input and prevent the propagation of a hard-to-find error that may pop up in other code that's calling this.

Reply:
ya I guess thats a good point, I could move this reduce out of the init and make the init accept the map instead ?

Reply:
Sure - at least that way we could verify that we aren't vulnerable to bad data as input for the class b/c the map can only have one value per token ID by definition...